x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/GroupingAggregatorEvaluationContext.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.aggregation;
+
+import org.elasticsearch.compute.data.BlockFactory;
+import org.elasticsearch.compute.operator.DriverContext;
+
+/**
+ * Provides a context for evaluating a grouping aggregator.
+ * A time-series aggregator might need more than just the DriverContext to evaluate, such as the range interval of each group.
+ */
+public class GroupingAggregatorEvaluationContext {
+    private final DriverContext driverContext;
```

Comment:
NIT: maybe this could be a record

Reply:
Ah, I made this a class intentionally so that we can extend for TimeSeriesGroupingAggregatorEvaluationContext.